From: Jinliang Zheng <alexjlzheng@tencent.com>
Subject: writeback: remove redundant checks for root memcg
Date: Tue, 8 Aug 2023 16:44:32 +0800

The check for root memcg will be done in wb_get_lookup(), so remove the
redundant one to simplify the code.

Link: https://lkml.kernel.org/r/20230808084431.1632934-1-alexjlzheng@tencent.com
Signed-off-by: Jinliang Zheng <alexjlzheng@tencent.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/backing-dev.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/mm/backing-dev.c~writeback-remove-redundant-checks-for-root-memcg
+++ a/mm/backing-dev.c
@@ -732,9 +732,6 @@ struct bdi_writeback *wb_get_create(stru
 
 	might_alloc(gfp);
 
-	if (!memcg_css->parent)
-		return &bdi->wb;
-
 	do {
 		wb = wb_get_lookup(bdi, memcg_css);
 	} while (!wb && !cgwb_create(bdi, memcg_css, gfp));
_
